request to fix mic numbers in polar plot indexing

falkben opened the pull request:
Had a question about where this comes from: 
data.mic_vh in the plot_bp_cross file.  It appears to be used to determine if channels are vertical or not...

That variable is loaded in during load_mic_info in the bp_proc.m file, but where does it originate from?

falkben committed:
Mic numbers shown in the polar plot were indexed using the unsorted list but plotted in the sorted list.

Left/right mixup?

leewujung commented:
I believe the current version is correct. If you look closely you will see that I sort the mic number in order to make the points on the polar plot connect in the right order (along azimuth angle, instead of along the mic number, which would result into criss-cross lines). The mic numbers are plotted with the sorted indices I think.

leewujung commented:
Also see the documentation for the mic_vh set up. It is an array users have to insert to indicate whether it's an element for the horizontal or vertical arrangement.

falkben commented:
The polar plots are sorted, but the numbering wasn't.

It looks like Anand needs to fill out the mic_vh for his horizontal array.

falkben committed:
added beam pattern on track view

falkben committed:
update track plot when changing items on bp_check_gui
fixed a bug with mic number labels on beampattern track plot

falkben committed:
Edit fields are disabled

falkben committed:
Fix text of mic numbers not ordered in cross/2D plot (elevation)

falkben committed:
view stable if rotated

falkben committed:
In cross configuration, center channel can be plotted as both hor. and az.  Changes in README.md and plotting to support this.

falkben committed:
Added example files

falkben committed:
Merge pull request #1 from falkben/bp_on_track

Bp on track

falkben committed:
Added options for stationary bat with head aim prescribed, bat to mic angle prescribed, and for setting the axis orientation to something other than [3,1,2].

falkben commented:
Looks like my latest commits are getting lumped into this old pull request.  I should have made a new branch for the stuff I did with Anand to avoid this.  In any case - today's commit is unrelated to the other work in this tree.

falkben committed:
Fixed a bug when calculating the channel alignment delays where if the recorded data had significant voltage bias, it could lead to incorrect cross correlations.

falkben committed:
Merge remote-tracking branch 'upstream/master'

falkben committed:
Updated README.md file to specify that directionality of the microphone is in degrees (not radians).

falkben committed:
Fixed conversion to Celsius

falkben committed:
refactored: passing data into bp_proc.m using data.params instead of multiple inputs
renamed typo variable tolernace to correct spelling tolerance
removed unmaintained batch_bp_proc.m files

falkben committed:
Added a bit of code for using the prescribed head direction even if you have 3 marker data.
Some code cleanup and better defaults on the batch_bp_proc.

falkben committed:
Added a button to quickly plot sequentially over a range of frequencies in order to check a call for it's response over it's full bandwidth.  At the moment, it only works as an "animation" for cross configuration plots, as the 2D plots take too long to plot.

falkben committed:
Added multi-frequency contour plot

Some refactoring and also sped up plotting of 2D data in the animation across freq

falkben committed:
Disable the bp_plot when in multifreq mode.

falkben committed:
Bug fixes

Clear the axes before switching plot types.  Make the axes not visible if not plotting on it.

falkben committed:
Changed some defaults, fixed bugs, cross multi-freq added

Made the default log (dB) scale
Made the multi-frequency plot for cross-config data plot in polar instead of 2D mode
Removed some "hold on" and "cla" calls that weren't needed.

falkben committed:
Load 'checked' file with bp_proc file

falkben committed:
Setting better defaults for batch processing

falkben committed:
Added a "follow peak freq" option to the check GUI

falkben committed:
Added PSD type specification
Power spectral density can either be FFT or pwelch
Fixed errors in how the peak to peak calculation was working

falkben committed:
Added dB RMS SPL calculation
Done with one third octave band filters at user specified freqs
Uses butterworth filter
bp_proc_check_GUI needs to be updated to allow option to select RMS instead of PSD

falkben committed:
Added dB RMS in the bp_check_GUI
Fixed a few bugs in save formatting for dB RMS
Fixed a few bugs in the edit call function in the GUI

falkben committed:
Compensate peak to peak and RMS for microphone directionality

falkben committed:
Compensate peak to peak and RMS for microphone directionality

falkben committed:
Merge branch 'master' of https://github.com/falkben/beampattern_processing

# Conflicts:
#	compensate_call_dB_fcn.m

leewujung commented on edit_call_section.m:
```diff
@@ -260,24 +260,63 @@ function reload_bp_timeseries(hh,mic_config)
-call_freq_vec = linspace(0,data.mic_data.fs/2,round((length(call_fft)+1)/2));
-call_fft_len = length(call_freq_vec);
-% call_psd = 2*abs(call_fft(1:call_fft_len)).^2/(length(call_fft)*data.mic_data.fs);
-call_psd = 2*abs(call_fft(1:call_fft_len)).^2;
```
In this version `call_psd` should really be `call_esd`, where psd is power spectral density and esd is energy spectral density.

leewujung commented on plot_bp_2d.m:
```diff
@@ -57,42 +67,58 @@ function plot_bp_2d(handles)
+  gridm('gcolor',190*ones(1,3)/255,'glinestyle','-');
+  axis off
+  if get(handles.checkbox_norm,'Value')==0  % if "normalized" not checked
+      geoshow(elqm/pi*180,azqm/pi*180,vq,'displaytype','texturemap');
```
Should update these to plotting projected bp intensity directly using projected x,y positions (from az,el), since geoshow is much slower than contourf

leewujung commented:
Checked all changes and merged into master.
